### DISCUSSION TITLE
Question: Is it possible to set custom styles on buttons in modals?

### DISCUSSION THREAD

**Discussion by nalnir** [mdx-editor/editor#298] [Q&A]
<img width="339" alt="Screenshot 2024-01-11 at 08 20 36" src="https://github.com/mdx-editor/editor/assets/28774583/14552add-80a7-4e2f-a5a4-47def1d7669c">
<img width="366" alt="Screenshot 2024-01-11 at 08 20 43" src="https://github.com/mdx-editor/editor/assets/28774583/fc2c6d8b-9c1a-4c7d-96df-07882ed22834">

It would nice to be able to change the button color, font, size etc. Is there a way to set custom styles for the modal component and the components inside it?

**Comment by braincore** [comment]
The general approach is to set the relevant css vars. See the dark theme docs: https://mdxeditor.dev/editor/docs/theming

For the button specifically, you can see the exact vars (font-size, color, ...) in the styles definition: https://github.com/mdx-editor/editor/blob/7d100d56ae7935c32f93bc1b48d49f1d6ea2184c/src/styles/ui.module.css#L460-L468

One oddity is that the button text color is set to `--baseBase` which is the same as the background color of a text input.

**Comment by pawelmidur** [comment]
Or you can overwrite the entire dialog component too

```
  ...
  imagePlugin({
    disableImageResize: true,
    ImageDialog: ImageDialog,
  }),
  ...
  ```
